sinaloa/src/sinaloa_nitro.rs

```diff
@@ -0,0 +1,351 @@
+                }
+            }
+
+            println!("SinaloaNitro::new native_attestation complete. instantiating Mexico City");
+            #[cfg(feature = "debug")]
+            let mexico_city_enclave = {
+                println!("Starting mexico city enclave in debug mode");
+                NitroEnclave::new(false, MEXICO_CITY_EIF_PATH, true, Some(SinaloaNitro::sinaloa_ocall_handler))
+                    .map_err(|err| SinaloaError::NitroError(err))?
+            };
+            #[cfg(not(feature = "debug"))]
+            let mexico_city_enclave = {
+                println!("Starting mexico city enclave in release mode");
+                NitroEnclave::new(false, MEXICO_CITY_EIF_PATH, false, Some(SinaloaNitro::sinaloa_ocall_handler))
+                    .map_err(|err| SinaloaError::NitroError(err))?
+            };
+            println!("SinaloaNitro::new NitroEnclave::new returned");
+            let meta = Self {
+                enclave: mexico_city_enclave,
+            };
+            println!("SinaloaNitro::new Mexico City instantiated. Calling initialize");
+            std::thread::sleep(std::time::Duration::from_millis(10000));
+
+            let initialize: MCMessage = MCMessage::Initialize(policy_json.to_string());
+
+            let encoded_buffer: Vec<u8> = bincode::serialize(&initialize)?;
+            meta.enclave.send_buffer(&encoded_buffer)?;
+
+            // read the response
+            let status_buffer = meta.enclave.receive_buffer()?;
+
+            let message: MCMessage = bincode::deserialize(&status_buffer[..])?;
+            let status = match message {
+                MCMessage::Status(status) => status,
+                _ => return Err(SinaloaError::MCMessageStatus(message)),
+            };
+            match status {
+                NitroStatus::Success => (),
+                _ => return Err(SinaloaError::NitroStatus(status)),
+            }
+            println!("SinaloaNitro::new complete. Returning");
+            return Ok(meta);
+        }
+
+        fn plaintext_data(&self, data: Vec<u8>) -> Result<Option<Vec<u8>>, SinaloaError> {
+            let parsed = colima::parse_mexico_city_request(&data)?;
+
+            if parsed.has_request_proxy_psa_attestation_token() {
+                let rpat = parsed.get_request_proxy_psa_attestation_token();
+                let challenge = colima::parse_request_proxy_psa_attestation_token(rpat);
+                let (psa_attestation_token, pubkey, device_id) =
+                    self.proxy_psa_attestation_get_token(challenge)?;
+                let serialized_pat = colima::serialize_proxy_psa_attestation_token(
+                    &psa_attestation_token,
+                    &pubkey,
+                    device_id,
+                )?;
+                Ok(Some(serialized_pat))
+            } else {
+                unreachable!("Unimplemented");
+            }
+        }
+
+        // Note: this function will go away
```

Comment:
This is copied from the Sinaloa trait isn't it? How long have these been marked "will go away"?

Reply:
Since the very beginning :-) 
They are functions that are useful for testing (Consider all of the tests that we have that don't exercise attestation), but if they were used in production, would be a security no-no.

I think it was valuable in the beginning to have tests that wouldn't get mired down in the attestation flows. They real question is, do we think we still need the "no_attestation" tests? Or would we be better off changing them to "with_attestation"? If we change them to "with_attestation", these functions can be removed from the whole code base (for all platforms), and we don't have to worry about users calling them and short-circuiting their security guarantees.

In any case, I think this is something that should be a separate issue/Merge Request instead of talking about it, doing it here.  I've opened issue #64

Reply:
Ah OK, I didn't realise that this was an old comment.  The diff made it look like it was something that had been added by this pull request.  I agree that this should be discussed outside of code review.